### PR TITLE
Add OpenVPN RW page

### PR DESCRIPTION
**Pull request by kevincela**
- Add OpenVPN RW page
- Add RW server management

RoadWarrior user managements will be covered in another PR

Card: https://trello.com/c/uJW5AAwp/178-openvpn-rw

Requires new backend: https://github.com/NethServer/nethsecurity/pull/247

### CODE REVIEW

File: public/i18n/en/translation.json [hunk 1]
```diff
@@ -1028,6 +1041,65 @@
     "vpn": {
       "title": "VPN"
     },
+    "openvpn_rw": {
+      "title": "OpenVPN Road Warrior",
+      "roadwarrior_server": "RoadWarrior server",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1046]:
```suggestion
      "roadwarrior_server": "Road Warrior server",
```

File: public/i18n/en/translation.json [hunk 2]
```diff
@@ -1028,6 +1041,65 @@
...
+      "rw_server": "RW server",
+      "roadwarrior_server_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1048]:
```suggestion
      "roadwarrior_server_description": "The OpenVPN Road Warrior is a VPN server designed for remote users, providing secure access to a private network from any internet location.",
```

File: public/i18n/en/translation.json [hunk 3]
```diff
@@ -1028,6 +1041,65 @@
...
+      "no_openvpn_rw_server_found": "No OpenVPN RoadWarrior server found",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1049]:
```suggestion
      "no_openvpn_rw_server_found": "No OpenVPN Road Warrior server found",
```

File: public/i18n/en/translation.json [hunk 4]
```diff
@@ -1028,6 +1041,65 @@
...
+      "custom_options": "Custom options",
+      "custom_options_tooltip": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+      "create": "Create",
+      "add_option": "Add option",
+      "delete_server_rw": "Delete server RW",
+      "delete_server_rw_message": "This action will also delete all account for this server.",
+      "database": "Database",
+      "status": "Status",
+      "connected_clients": "Connected clients",
+      "roadwarrior_accounts": "RoadWarrior accounts",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1090]:
```suggestion
      "roadwarrior_accounts": "Road Warrior accounts",
```

File: public/i18n/en/translation.json [hunk 5]
```diff
@@ -1028,6 +1041,65 @@
...
+      "username_password": "Username and password",
+      "certificate": "Certificate",
+      "username_password_certificate": "Username, password and certificate",
+      "username_otp_certificate": "Username and OTP certificate",
+      "local_database": "Local database",
+      "custom_option_combobox_placeholder": "Search key",
+      "strong": "Strong",
+      "weak": "Weak",
+      "server_configured": "Server configured",
+      "roadwarrior_server_configured_successfully": "RoadWarrior server configured successfully"
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1101]:
```suggestion
      "roadwarrior_server_configured_successfully": "Road Warrior server configured successfully"
```

File: public/i18n/en/translation.json [hunk 6]
```diff
@@ -1028,6 +1041,65 @@
...
+      "server_name": "Server name",
+      "user_database": "User database",
+      "create_accounts_for_all_users": "Create account for all users",
+      "create_accounts_for_all_users_tooltip": "All accounts created will have a certificate valid for 3650 days.",
+      "authentication_mode": "Authentication mode",
+      "mode": "Mode",
+      "routed": "Routed",
+      "bridged": "Bridged",
+      "vpn_network": "VPN network",
+      "vpn_network_tooltip": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1061]:
```suggestion
      "vpn_network_tooltip": "The VPN network is a virtual network used by clients; please select a network not used in any other firewall configuration.",
```

File: public/i18n/en/translation.json [hunk 7]
```diff
@@ -1028,6 +1041,65 @@
...
+      "bridge": "Bridge",
+      "range_ip_start": "Range IP start",
+      "range_ip_end": "Range IP end",
+      "add_network": "Add network",
+      "allow_client_to_client_network_traffic": "Allow client-to-client network traffic",
+      "compression": "Compression",
+      "digest": "Digest",
+      "cipher": "Cipher",
+      "enforce_minimum_tls_version": "Enforce a minimum TLS version",
+      "custom_options": "Custom options",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1081]:
```suggestion
      "custom_options": "Custom DHCP options",
```

File: public/i18n/en/translation.json [hunk 8]
```diff
@@ -1028,6 +1041,65 @@
...
+      "custom_options_tooltip": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1082]:
```suggestion
      "custom_options_tooltip": "Send additional DHCP options to clients; especially useful for configuring an OpenVPN client to connect to a Windows network over the VPN",
```

File: public/i18n/en/translation.json [hunk 9]
```diff
@@ -1028,6 +1041,65 @@
...
+      "create": "Create",
+      "add_option": "Add option",
+      "delete_server_rw": "Delete server RW",
+      "delete_server_rw_message": "This action will also delete all account for this server.",
+      "database": "Database",
+      "status": "Status",
+      "connected_clients": "Connected clients",
+      "roadwarrior_accounts": "RoadWarrior accounts",
+      "roadwarrior_accounts_description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1091]:
```suggestion
      "roadwarrior_accounts_description": "Users in this list can connect to the OpenVPN Road Warrior server. Only users with a configured password can use the 'username and password' authentication mode. If the selected authentication mode requires an x509 certificate, the certificate must be valid.",
```

**Review by gsanchietti — COMMENTED** [review]
Replaced placeholder and fixed the name of the page: it should be `OpenVPN Road Warrior` everywhere

File: src/components/standalone/openvpn_rw/CreateOrEditRWServerDrawer.vue [hunk 10]
```diff
@@ -0,0 +1,646 @@
...
+const authMode = ref('username_password')
+const mode = ref('routed')
+const vpnNetwork = ref('')
+const publicIpOrHostname = ref<string[]>([])
+const protocol = ref('udp')
+const port = ref('')
+const routeTrafficThroughVpn = ref(false)
+const pushCustomNetworkRoutes = ref<string[]>([])
+const clientToClientNetworkTraffic = ref(false)
+const compression = ref('disabled')
```
Comment (gsanchietti) [inline src/components/standalone/openvpn_rw/CreateOrEditRWServerDrawer.vue:73]: Value for disabled is an empty string:
```suggestion
const compression = ref('')
```

File: src/components/standalone/openvpn_rw/CreateOrEditRWServerDrawer.vue [hunk 11]
```diff
@@ -0,0 +1,646 @@
...
+const digestOptions = ref<NeComboboxOption[]>([])
+const cipherOptions = ref<NeComboboxOption[]>([])
+const authModeOptions = ref<NeComboboxOption[]>([])
+const bridgesOptions = ref<NeComboboxOption[]>([])
+const userDatabaseOptions = ref<NeComboboxOption[]>([])
+const availableCustomOptions = ref<NeComboboxOption[]>([])
+
+const compressionOptions = [
+  {
+    id: 'disabled',
```
Comment (gsanchietti) [inline src/components/standalone/openvpn_rw/CreateOrEditRWServerDrawer.vue:91]:
```suggestion
    id: '',
```

**Review by gsanchietti — COMMENTED** [review]
It seems that the `disabled` value for compress is not valid.

File: src/components/standalone/openvpn_rw/RWServerDetails.vue [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+      </div>
+      <div
+        class="ml-4 mr-10 grow border-l border-gray-800 py-3 pl-4 text-sm dark:border-gray-600 md:ml-8 md:pl-8"
+      >
+        <div class="flex max-w-xl flex-col justify-between gap-y-4 sm:flex-row sm:gap-x-4">
+          <div>
+            <p class="mb-2 font-semibold">{{ t('standalone.openvpn_rw.status') }}:</p>
+            <div :class="['flex', 'flex-row', 'items-center']">
+              <font-awesome-icon
+                :icon="['fas', true ? 'circle-check' : 'circle-xmark']"
```
Comment (gsanchietti) [inline src/components/standalone/openvpn_rw/RWServerDetails.vue:42]: The icon is always green, even if the server is disabled.
```suggestion
                :icon="['fas', server.enabled == '1' ? 'circle-check' : 'circle-xmark']"
```

File: public/i18n/en/translation.json (hunk 5 above)
Comment (kevincela) [reply public/i18n/en/translation.json]: This item was directly removed after talking with Andrea, we'll be using a single title for the toast instead

File: src/components/standalone/openvpn_rw/CreateOrEditRWServerDrawer.vue (hunk 10 above)
Comment (kevincela) [reply src/components/standalone/openvpn_rw/CreateOrEditRWServerDrawer.vue:73]: For this issue i've simply changed line 363 in the drawer component, i forgot to change the correct value to associate with the default state

**Review by Tbaile — APPROVED** [review]

File: public/i18n/en/translation.json [hunk 13]
```diff
@@ -1028,6 +1041,62 @@
...
+      "custom_dhcp_options_tooltip": "Send additional DHCP options to clients; especially useful for configuring an OpenVPN client to connect to a Windows network over the VPN",
+      "create": "Create",
+      "add_option": "Add option",
+      "delete_roadwarrior_server": "Delete Road Warrior server",
+      "delete_roadwarrior_server_message": "This action will also delete all accounts for this server.",
+      "database": "Database",
+      "status": "Status",
+      "connected_clients": "Connected clients",
+      "roadwarrior_accounts": "Road Warrior accounts",
+      "roadwarrior_accounts_description": "Users in this list can connect to the OpenVPN Road Warrior server. Only users with a configured password can use the 'username and password' authentication mode. If the selected authentication mode requires an x509 certificate, the certificate must be valid.",
```
Comment (gsanchietti) [inline public/i18n/en/translation.json:1090]:
```suggestion
      "roadwarrior_accounts_description": "Users in this list can connect to the OpenVPN Road Warrior server. Only users with a configured password can use the 'username and password' authentication mode. If the selected authentication mode requires an x509 certificate, the certificate must be valid. Changes to users take effect immediately.",
```

**Review by andre8244 — APPROVED** [review]

**Pull request merged**